icu4c/source/common/cmemory.h

```diff
@@ -101,12 +101,18 @@ typedef union {
  * Create & return an instance of "type" in statically allocated storage.
  * e.g.
  *    static std::mutex *myMutex = STATIC_NEW(std::mutex);
- * This is intended for use when
+ * To destroy an object created in this way, invoke the destructor explicitly, e.g.
+ *    myMutex->~mutex();
+ * DO NOT use delete.
+ * DO NOT use with class UMutex, which has specific support for static instances.
+ *
+ * STATIC_NEW is intended for use when
  *   - We want a static (or global) object.
- *   - We don't want it to ever be destructed, to avoid order-of-destruction
- *     or use-after-destruction problems.
- *   - We want to avoid an ordinary heap allocated object, to avoid triggering
- *     memory leak reports, from valgrind, for example.
```

Comment:
Thanks Andy!
If this is merged, then there are a few other places in the ICU code that could likely use this new macro as well. :)